### PR TITLE
Authorisation flow #1 - login

### PR DESCRIPTION
**Pull request by wojciech-bilicki**

### CODE REVIEW

File: picby_graphql_api/src/index.ts [hunk 1]
```diff
@@ -15,12 +20,38 @@ import { RegisterResolver } from './modules/user/Register';
...
+        client: redis as any
+      }),
+      name: 'qid',
+      secret: 'asdasda',
+      resave: false,
+      saveUninitialized: false,
+      cookie: {
+        httpOnly: true,
+        secure: process.env.NODE_ENV === 'production',
+        maxAge: 1000*60*60*24*7*365 //7 years
```
Comment (pau-gumienna) [inline picby_graphql_api/src/index.ts]: magic number

File: picby_graphql_api/src/index.ts [hunk 2]
```diff
@@ -15,12 +20,38 @@ import { RegisterResolver } from './modules/user/Register';
...
+    origin: '*'
+  }))
+  
+  app.use(
+    session({
+      store: new RedisStore({
+        client: redis as any
+      }),
+      name: 'qid',
+      secret: 'asdasda',
```
Comment (pau-gumienna) [inline picby_graphql_api/src/index.ts]: we probably should set variables at the top of the file, test ones too
+ more a tip than a comment but i think we should take care of the import order, at the top libs then our files

File: picby_graphql_api/src/modules/user/Login.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { ValidationError } from 'apollo-server-core';
+import * as bcrypt from 'bcryptjs';
+import { Context } from 'src/types/Context';
+import { Arg, Ctx, Mutation, Resolver } from "type-graphql";
+import { User } from '../../entity/User';
+
+
+
```
Comment (v4sp) [inline picby_graphql_api/src/modules/user/Login.ts]: \n's

File: picby_graphql_api/src/modules/user/Login.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    @Arg("email") email: string,
+    @Arg("password") password: string,
+    @Ctx() ctx: Context
+  ):Promise<User | null> {
+    const user = await User.findOne({where: {email}})
+    if(!user) {
+      return null;
+    }
+
+
```
Comment (v4sp) [inline picby_graphql_api/src/modules/user/Login.ts]: \n's

File: picby_graphql_api/src/index.ts [hunk 5]
```diff
@@ -15,12 +20,38 @@ import { RegisterResolver } from './modules/user/Register';
...
+        httpOnly: true,
+        secure: process.env.NODE_ENV === 'production',
+        maxAge: 1000*60*60*24*7*365 //7 years
+
+      }
+    })
+  )
+
   apolloServer.applyMiddleware({ app, cors: false });
   const port = process.env.PORT || 4000;
```
Comment (v4sp) [inline picby_graphql_api/src/index.ts]: could declare 4000 as defaultPort or w/e

**Pull request merged**